src/app/components/multi-page-form/multi-page-form.js

```diff
@@ -1,138 +1,167 @@
-            </button>
-            <button
-                type="submit" className="primary"
-                hidden={currentPage < pages}
-                onClick={validateAndSubmit}
-                disabled={disabled}
-            >
-                Submit
-            </button>
-            <button
-                type="button" className="primary next"
-                hidden={currentPage === pages}
-                onClick={nextPage}
-                disabled={disabled}
-            >
-                Next
-            </button>
+            <BackButton disabled={disabled} />
+            <SubmitButton
+                disabled={disabled} onSubmit={onSubmit}
+                formRef={formRef}
+            />
+            <NextButton disabled={disabled} />
         </div>
     );
 }
 
-export default function MultiPageForm({
+function MultiPageFormInContext({
     children,
     validatePage=pass, onPageChange=pass, onSubmit=pass,
     submitting, ...formParams
 }) {
-    const [currentPage, setCurrentPage] = useState(1);
-    const [validatedPages, setValidatedPages] = useState({});
-    const formRef = useRef();
-    const activeRef = useRef();
-    const pages = children.length;
-    const buttonRowProps = {
-        pages,
-        currentPage,
-        setCurrentPage,
-        validatePage,
-        validatedPages,
-        setValidatedPages,
-        activeRef,
-        formRef,
-        onSubmit
-    };
-
-    useEffect(() => {
-        onPageChange(currentPage);
-    }, [currentPage, onPageChange]);
+    const formRef = React.useRef();
 
     return (
         <div className="multi-page-form">
             <form
                 acceptCharset="UTF-8" className="form"
-                method="post" ref={formRef} {...formParams}>
-                <MarkupChildren
-                    children={children}
-                    currentPage={currentPage}
-                    activeRef={activeRef}
-                    validatedPages={validatedPages}
-                    onSubmit={onSubmit}
-                />
-                <PageCount {...{pages, currentPage}} />
-                {
-                    submitting &&
-                        <div className="big-message">Submitting...</div>
-                }
-                <ButtonRow disabled={submitting} {...buttonRowProps} />
+                method="post" ref={formRef} {...formParams}
+            >
+                <PagesContextProvider
+                    contextValueParameters={{
+                        pages: children.length,
+                        validatePage,
+                        onPageChange
+                    }}
+                >
+                    <MarkupChildren children={children} />
+                    <PageCount />
+                    {
+                        submitting &&
+                            <div className="big-message">Submitting...</div>
+                    }
+                    <ButtonRow disabled={submitting} {...{formRef, onSubmit}} />
```

Comment:
BTW, I wrote a fancy "pendingState" hook for Kinetic that waits for 150ms to display.  the concept behind that is that you don't want a spinner to display for very short requests because it causes flickering.  Our [LoadingAnimation](https://github.com/openstax/kinetic/blob/main/frontend/src/components/loading-animation.tsx#L185) uses it internally.  Maybe it'll be of use to you

https://github.com/openstax/kinetic/blob/3d0c1de92383365a8fb5a699e2180dfcadd7c2be/frontend/src/lib/hooks.ts#L4-L19